modules/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java

```diff
@@ -347,6 +367,11 @@ public void visit(BLangTypeCastExpr castExpr) {
         BType targetType = symResolver.resolveTypeNode(castExpr.typeNode, env);
         BType sourceType = checkExpr(castExpr.expr, env, Lists.of(symTable.noType)).get(0);
 
+        if (sourceType == symTable.errType || targetType == symTable.errType) {
```

Comment:
Can you please explain the reason for this condition?

Reply:
Remaining TypeCast logic fails if the source or target type is set to err.type. Hence added this validation.